ReactiveCocoa/Swift/Property.swift

```diff
@@ -171,11 +226,21 @@ public final class MutableProperty<Value>: MutablePropertyType {
 	public init(object: NSObject?, keyPath: String) {
 		self.object = object
 		self.keyPath = keyPath
-		
+
 		/// DynamicProperty stay alive as long as object is alive.
 		/// This is made possible by strong reference cycles.
 		super.init()
-		object?.rac_willDeallocSignal()?.toSignalProducer().startWithCompleted { self }
+
+		object?.rac_valuesForKeyPath(keyPath, observer: nil)?
+			.toSignalProducer()
+			.start {
+				switch $0 {
+				case let .Next(newValue):
+					self._property?.value = newValue
+				default:
+					self._property = nil
```

Comment:
Also, can you use an explicit list of events instead of `default`? If we add a new event this would force us to think about how to handle it, whereas `default` would ignore that fact.